crates/orchestrator/src/tests/jobs/da_job/mod.rs

```diff
@@ -1,79 +1,106 @@
+use crate::{config::config, jobs::Job};
+use assert_matches::assert_matches;
+use color_eyre::eyre::eyre;
+use da_client_interface::MockDaClient;
+use mockall::predicate::always;
+use rstest::rstest;
+use serde_json::json;
+use starknet_core::types::{FieldElement, MaybePendingStateUpdate, PendingStateUpdate, StateDiff};
+use std::collections::HashMap;
+use uuid::Uuid;
 
+/// Tests the DA Job's handling of a blob length exceeding the supported size.
+/// It mocks the DA client to simulate the environment and expects an error on job processing.
+/// Validates the error message for exceeding blob limits against the expected output.
+/// Asserts correct behavior by comparing the received and expected error messages.
 #[rstest]
+#[case(
+    "src/tests/jobs/da_job/test_data/state_update/638353.txt",
+    "src/tests/jobs/da_job/test_data/nonces/638353.txt",
+    "63853",
+    110
+)]
 #[tokio::test]
-async fn test_verify_job(#[from(default_job_item)] mut job_item: JobItem) {
+async fn test_da_job_process_job_failure_on_small_blob_size(
+    #[case] state_update_file: String,
+    #[case] nonces_file: String,
+    #[case] internal_id: String,
+    #[case] current_blob_length: u64,
+) {
+    // Mocking DA client calls
     let mut da_client = MockDaClient::new();
-    da_client.expect_verify_inclusion().times(1).returning(|_| Ok(DaVerificationStatus::Verified));
+    // dummy state will have more than 1200 bytes
+    da_client.expect_max_blob_per_txn().with().returning(|| 1);
+    da_client.expect_max_bytes_per_blob().with().returning(|| 1200);
+
+    let server = TestConfigBuilder::new().mock_da_client(Box::new(da_client)).build().await;
+    let config = config().await;
+
+    let state_update = read_state_update_from_file(state_update_file.as_str()).expect("issue while reading");
+
+    let state_update = MaybePendingStateUpdate::Update(state_update);
+    let state_update = serde_json::to_value(&state_update).unwrap();
+    let response = json!({ "id": 640641,"jsonrpc":"2.0","result": state_update });
+
+    get_nonce_attached(&server, nonces_file.as_str());
+
+    let state_update_mock = server.mock(|when, then| {
+        when.path("/").body_contains("starknet_getStateUpdate");
+        then.status(200).body(serde_json::to_vec(&response).unwrap());
+    });
+
+    let max_blob_per_txn = config.da_client().max_blob_per_txn().await;
+
+    let response = DaJob
+        .process_job(
+            config.as_ref(),
+            &mut JobItem {
+                id: Uuid::default(),
+                internal_id: internal_id.to_string(),
+                job_type: JobType::DataSubmission,
+                status: JobStatus::Created,
+                external_id: ExternalId::String(internal_id.to_string().into_boxed_str()),
+                metadata: HashMap::default(),
+                version: 0,
+            },
+        )
+        .await;
+
+    assert_matches!(response,
+        Err(e) => {
+            let expected_error = eyre!(
```

Comment:
could be worth having it's own macro like `assert_eyre_matches`

Reply:
or is the plan to move asap to real errors ?

Reply:
We indeed have plans to move to `thisError` for all error cases.
Currently it is implemented in a separate PR : #66 
( Note that the PR #66 is accidentally closed, and will be included in after PR #51 (this one) is merged, this is done to ensure `separation of concern`. )




### IMPORTANT 
[This branch is a fork from [increasing coverage](https://github.com/madara-alliance/madara-orchestrator/tree/feat/increasing-test-coverage) ]
[This comment refers to changed already merged to main, they are shown here since #52 is merged and the base of this PR automatically shifted to `main` ]